src/qibo/quantum_info/utils.py

```diff
@@ -339,6 +339,68 @@ def hellinger_shot_error(
     return hellinger_error
 
 
+def total_variation_distance(
+    prob_dist_p, prob_dist_q, validate: bool = False, backend=None
+):
+    """Calculates the total variation distance between two discrete probability distributions.
+
+    For probabilities :math:`p` and :math:`q`, the total variation distance is defined as
+
+    .. math::
+        \\operatorname{TVD}(p, q) = \frac{1}{2} \\, \\norm{p - q}_{1}
+            = \\frac{1}{2} \\, \\sum_{x} \\, \\abs{p(x) - q(x)} \\, ,
+
+    where :math:`\\norm{\\cdot}_{1}` detones the :math:`\\ell_{1}`-norm.
+
+    Args:
+        prob_dist_p (ndarray or list): discrete probability distribution :math:`p`.
+        prob_dist_q (ndarray or list): discrete probability distribution :math:`q`.
+        validate (bool, optional): if ``True``, checks if :math:`p` and :math:`q` are proper
+            probability distributions. Defaults to ``False``.
+        backend (:class:`qibo.backends.abstract.Backend`, optional): backend to be
+            used in the execution. If ``None``, it uses
+            :class:`qibo.backends.GlobalBackend`. Defaults to ``None``.
+
+    Returns:
+        float: Total variation distance measure.
+    """
+    backend = _check_backend(backend)
+
+    if isinstance(prob_dist_p, list):
+        prob_dist_p = backend.cast(prob_dist_p, dtype=np.float64)
+
+    if isinstance(prob_dist_q, list):
+        prob_dist_q = backend.cast(prob_dist_q, dtype=np.float64)
```

Comment:
I don't know whether the cast is needed in the validation process, but `backend.calculate_norm` is casting already by default.

Reply:
It is needed because the checks are using the `shape` attribute that lists do not have